### PR TITLE
Minor tweaks in ThingFactory and ModuleFactory classes

### PR DESCRIPTION
**Pull request by xezon**
A few tweaks.

### CODE REVIEW

File: src/game/common/modules/modulefactory.cpp [hunk 1]
```diff
@@ -79,6 +79,7 @@ const ModuleFactory::ModuleTemplate *ModuleFactory::Find_Module_Template(const U
         return &it->second;
     }
 
+    captainslog_dbgassert(0, "Module name '%s' not found", name.Str());
```
Comment (xezon) [inline src/game/common/modules/modulefactory.cpp:82]: I saw this in one of the binaries.

File: src/game/common/thing/thingfactory.cpp [hunk 2]
```diff
@@ -52,9 +52,7 @@ ThingTemplate *ThingFactory::New_Template(const Utf8String &name)
     new_template->Friend_Set_Template_ID(m_nextTemplateID);
     m_nextTemplateID++;
 
-    if (m_nextTemplateID == 0) {
-        captainslog_dbgassert(0, "m_nextTemplateID wrapped to zero");
-    }
+    captainslog_dbgassert(m_nextTemplateID != 0, "m_nextTemplateID wrapped to zero");
```
Comment (xezon) [inline src/game/common/thing/thingfactory.cpp:55]: Condition can be part of assert itself.

File: src/game/common/thing/thingfactory.cpp [hunk 3]
```diff
@@ -68,12 +66,12 @@ ThingTemplate *ThingFactory::New_Override(ThingTemplate *thing_template)
         "Thing template '%s' not in master list",
         thing_template->Get_Name().Str());
 
+    ThingTemplate *override_template = static_cast<ThingTemplate *>(thing_template->Friend_Get_Final_Override());
```
Comment (xezon) [inline src/game/common/thing/thingfactory.cpp:69]: Looks like it was `Friend_Get_Final_Override`, which is same as `Get_Final_Override` though. But it allows for `static_cast` here, which is nicer.

File: src/game/common/thing/thingfactory.cpp [hunk 4]
```diff
@@ -217,15 +215,15 @@ void ThingFactory::Parse_Object_Definition(INI *ini, Utf8String name, Utf8String
     }
 
     if (reskin_from.Is_Not_Empty()) {
-        ThingTemplate *that = g_theThingFactory->Find_Template(reskin_from, true);
+        ThingTemplate *that = g_theThingFactory->Find_Template_Internal(reskin_from, true);
```
Comment (xezon) [inline src/game/common/thing/thingfactory.cpp]: Non-World Builder shows `Find_Template_Internal` here, which looks more logical too.

Comment (jonwil) [reply src/game/common/thing/thingfactory.cpp]: WorldBuilder confirms the original code used Find_Template. Find_Template just happens to be inlined into the other binaries.

**Review by jonwil — CHANGES_REQUESTED** [review]
Other than the Find_Template thing, this looks good to me.

File: src/game/common/thing/thingfactory.cpp (hunk 4 above)
Comment (xezon) [reply src/game/common/thing/thingfactory.cpp]: reverted.

**Review by jonwil — APPROVED** [review]
Looks good to me now. Approved.

**Review by tomsons26 — APPROVED** [review]

**Pull request merged**